limit user input to extra fields for certain item types

MarcelBolten opened the pull request:
Closes #3984

If `{"eLabFTW_hide_body": true}` is set in the metadata root, an entities body (experiment or items) will not be displayed in view and edit mode via css: `display:none;`. css is used to avoid first showing the body and than hiding it after JS is executed.
In show mode the metadata will be shown instead of the body if the 'toggle content' button is pressed.

NicolasCARPi reviewed (commented):
I think it might be a good opportunity to start a namespaced key. Also, I'd vote for all lowercase keys. So it could be:

~~~json
{
"elabftw" : [
    "display_main_text": false
    ]
}
~~~

Because there is also the question of: hide_body or display_main_text. I think it's easier to reason with positives options. And `display_main_text` => `false` looks clearer to me.

We could also allow setting the extra_fields in there, it shouldn't be too much to look for it both in root namespace and in elabftw namespace. Then maybe add a warning somewhere with a button to fix it, should be easy to do, too.

MarcelBolten commented:
I added the `elabftw` namespace and renamed the key `eLabFTW_hide_body` to `display_main_text` as requested.
```json
{
  "elabftw": {
    "display_main_text": false
  }
}
```
Note that I did not use an array but an object.

> We could also allow setting the extra_fields in there, it shouldn't be too much to look for it both in root namespace and in elabftw namespace. Then maybe add a warning somewhere with a button to fix it, should be easy to do, too.

Yes, sounds good but I would like to open a new issus/PR for this additional part.

NicolasCARPi commented:
Great, thanks!

The way I see it, for now we support extra_fields both at root and in elabftw namespace. Then we can promote in the doc and the [builder](https://github.com/elabftw/elabftw/discussions/3990) putting it in in elabftw namespace. Simply display a warning if extra_fields is found at root, with a button to fix it (pretty simple). Then one day we can do an update where all metadata that contain only an extra_fields at root will get namespaced. But this is for later and you are right to limit the scope of this PR, it is also my idea.

NicolasCARPi reviewed (approved):
Thank you! I've made a few fixes. Also hide the "Save" button because it only saves the body.

MarcelBolten commented:
> The way I see it, for now we support extra_fields both at root and in elabftw namespace.

I started looking into it and have a first version working.

> Simply display a warning if extra_fields is found at root, with a button to fix it (pretty simple).

This is next.
